server/routes/auth.js

```diff
@@ -0,0 +1,57 @@
+const Router = require('koa-router');
+const User = require('../models/user');
+const validateAuthRoutes = require('../middleware/validateAuthRoutes')
+const bcrypt = require('bcrypt');
+const getUserByUsername = require('../middleware/authenticate');
+const jsonwebtoken = require('jsonwebtoken');
+const jwt = require('../middleware/jwt');
+
+const router = new Router({
+    prefix: '/auth'
+});
+
+
+/**
+ * You can register with:
+ * curl -X POST -d 'user[email]=kenya@tunapanda.org&user[username]=mountkenya&user[hash]=mountkenya' http://localhost:3000/api/v1/auth/register
+ * curl -X POST -d '{ "user": { "username": "okemwa", "hash": "okemwaokewamaokemwa", "email":"moses@okemwa.org" } }' http://localhost:3000/api/v1/auth/register
+ */
+router.post('/register', validateAuthRoutes.validateNewUser, getUserByUsername, async ctx => {
```

Comment:
Although it is authentication related, in order to maintain the CRUD convention, user creations should be via POST to `/users`

Reply:
Mick made these changes. They will be included in the next push to this branch